Code/Robot/Robot/Robot.ino

```diff
@@ -21,28 +21,56 @@ int main() {
 }
 
 void loop() {
-  bool start;
+  bool start_signal;
   states state = DRIVING;
+  String inputString = "";            // Holds incoming data from Serial3
+  boolean messageInProgress = false;  // Indicates if message parsing is in progress
   while (true) {
-    // Check if data is available to read from Serial3
-    if (Serial3.available() > 0) {  // zero will change here so I'm not going to try and predict what it will be
-      // Read the incoming byte from Serial3
-      char incomingByte = Serial3.read();
-
-      // Check if the incoming byte is '1' or '0' and set the start flag accordingly
-      if (incomingByte == '1') {
-        start = true;
-      } else if (incomingByte == '0') {
-        start = false;
-      }
+
+    String message = read_serial(inputString, messageInProgress);
+    
+    // Check if the incoming byte is '1' or '0' and set the start flag accordingly
+    if (message == "") {
+
+    } else if (message == "<1>") {
```

Comment:
I wonder if there is a library to serialize standard data structures in arduino c. Would be very nice to be able to send a list of occupied spots on the chassis and parse it directly as an array. 

If not, we can surely write our own array serialization